pages/_app.js

```diff
@@ -92,8 +99,15 @@ export default function App({ Component, pageProps }) {
   function handleResetFilter() {
     setFilters({ lightNeed: null, waterNeed: null, fertiliserSeason: [] });
     setIsFilterVisible(false);
+    setSearchQuery("");
   }
 
+  router.events?.on("routeChangeStart", (url) => {
+    if (url === "/myplants" || url === "/") {
+      handleResetFilter();
+    }
+  });
+
   // FYI: For better usability, we deviate from our user story and leave only one form expanded at a time
```

Comment:
Comments should be removed, especially if they appear to be outdated like this one.